From a887a1577f6ca6003141a75084f5d6c2c8019200 Mon Sep 17 00:00:00 2001
From: jjian zhou <jjian.zhou@mediatek.com>
Date: Fri, 26 Apr 2019 11:12:06 +0800
Subject: [PATCH 7/7] CHROMIUM: arm64: dts: mt8183-kukui: add wifi power
 controller

To comply with schematic design, remove gpio119 from
mmc1_fixed_power and add it back as wlan_en pin in
mmc1default. msdcpll is only for mmc module. Assign
msdcpll to mmc1 clk for avoiding unexpected changes.

BUG=b:131043339
TEST=Cherry-pick and update kernel with
     https://crrev.com/c/1585667. Make sure SDIO clk
     is running at 200MHz correctly by checking
     /sys/kernel/debug/mmc1/ios

Change-Id: I325df93ae939e929618f7234f18c674cce4e4138
Signed-off-by: jjian zhou <jjian.zhou@mediatek.com>
Signed-off-by: Ayo Wu <ayowu@chromium.org>
---
 .../arm64/boot/dts/mediatek/mt8183-kukui.dtsi | 22 ++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi b/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi
index 0bf7b3c081bf..21f30d0e00b5 100644
--- a/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi
@@ -68,6 +68,20 @@
 		enable-active-high;
 	};
 
+	mmc1_fixed_power: regulator@3 {
+		compatible = "regulator-fixed";
+		regulator-name = "mmc1_power";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	mmc1_fixed_io: regulator@4 {
+		compatible = "regulator-fixed";
+		regulator-name = "mmc1_io";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+	};
+
 	reserved_memory: reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
@@ -392,8 +406,8 @@
 	pinctrl-names = "default", "state_uhs";
 	pinctrl-0 = <&mmc1_pins_default>;
 	pinctrl-1 = <&mmc1_pins_uhs>;
-	vmmc-supply = <&mt6358_vmch_reg>;
-	vqmmc-supply = <&mt6358_vmc_reg>;
+	vmmc-supply = <&mmc1_fixed_power>;
+	vqmmc-supply = <&mmc1_fixed_io>;
 	bus-width = <4>;
 	max-frequency = <200000000>;
 	drv-type = <2>;
@@ -406,6 +420,8 @@
 	non-removable;
 	no-mmc;
 	no-sd;
+	assigned-clocks = <&topckgen CLK_TOP_MUX_MSDC30_1>;
+	assigned-clock-parents = <&topckgen CLK_TOP_MSDCPLL_D2>;
 };
 
 &mt6358_vdram2_reg {
@@ -716,7 +732,7 @@
 			mediatek,pull-down-adv = <10>;
 		};
 
-		pins_wifi {
+		pins_wlan_en {
 			pinmux = <PINMUX_GPIO119__FUNC_GPIO119>;
 			output-high;
 		};
-- 
2.21.0.1020.gf2820cf01a-goog

